independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/BeanGenerator.java

```diff
@@ -96,28 +96,35 @@ public BeanGenerator(AnnotationLiteralProcessor annotationLiterals, Predicate<Do
     /**
      *
      * @param bean
+     * @param existingClasses
```

Comment:
So I think that we could set both the `existingClasses` and the map to the `BeanGenerator` final fields and avoid passing more arguments in all those methods. The same applies to `ReflectionRegistration` BTW. @stuartwdouglas I can add a commit to your PR if you don't mind ;-).

Reply:
Done :-)